lib/src/equals_matcher.dart

```diff
@@ -260,16 +269,48 @@ class _DeepMatcher extends Matcher {
   @override
   Description describeMismatch(
       item, Description mismatchDescription, Map matchState, bool verbose) {
-    var reason = matchState['reason'] as String ?? '';
-    // If we didn't get a good reason, that would normally be a
-    // simple 'is <value>' message. We only add that if the mismatch
-    // description is non empty (so we are supplementing the mismatch
-    // description).
-    if (reason.isEmpty && mismatchDescription.length > 0) {
-      mismatchDescription.add('is ').addDescriptionOf(item);
+    var mismatch = matchState['mismatch'] as _Mismatch;
+    if (mismatch.location.isNotEmpty) {
+      mismatchDescription
+          .add('at location ')
+          .add(mismatch.location)
+          .add(' is ')
+          .addDescriptionOf(mismatch.actual);
+      if (mismatch.describeProblem != null) {
+        mismatchDescription
+            .add(' ${mismatch.instead ? 'instead of ' : 'which'} ');
```

Comment:
This looks like where the extra space is coming from.

```suggestion
            .add(' ${mismatch.instead ? 'instead of' : 'which'} ');
```

It could be that this string isn't currently covered by any tests here?

Reply:
Fixed.  It turns out that this string *is* covered by tests, it’s just that the error messages are all matched ignoring whitespace (see `shouldFail` in `test/test_utils.dart`).  It seems a little silly to add a test just to make sure this whitespace issue doesn’t regress, but I can do so if you want.